### PR TITLE
Add admin users

### PR DESCRIPTION
**Pull request by Unesdala**
-Added initial form logic
-Added error messaging (Needs to be moved to separate API, as there's no way to clear and re-enable button atm)
-Disables button if errors

### CODE REVIEW

**Comment by Unesdala** [comment]
@JoshuaVSherman  can you take a look at the last test for AdminController and fix the issues with verifying the credentials for userRoles.indexOf(r.body.userType)?

**Comment by JoshuaVSherman** [comment]
@Unesdala if this is ready, please remove WIP from title, thanks

File: src/containers/AdminDashboard/AdminController.tsx [hunk 1]
```diff
@@ -243,5 +245,84 @@ class AdminController {
...
+        });
+    // eslint-disable-next-line no-console
+    } catch (e) { console.log(e); return false; }
+    if (r.status === 400) {
+      // Create user. Include userType for admin for clc (clc-admin)
+      this.view.setState({ formError: '' });
+      return true;
+    }
+    if (r.body._id && userRoles.indexOf(r.body.userType) === -1) {
+      // if user exists, and isn't an admin, update by user email. Set userType = clc-admin
```
Comment (JoshuaVSherman) [inline src/containers/AdminDashboard/AdminController.tsx:319]: @Unesdala please remove mention of userType here, should not be in our Github, thanks

File: src/containers/AdminDashboard/AdminController.tsx [hunk 2]
```diff
@@ -243,5 +245,84 @@ class AdminController {
...
+    try {
+      r = await this.superagent.post(`${process.env.BackendUrl}/user`)
+        .set('Authorization', `Bearer ${auth.token}`)
+        .set('Accept', 'application/json')
+        .send({
+          email: addAdminEmail,
+        });
+    // eslint-disable-next-line no-console
+    } catch (e) { console.log(e); return false; }
+    if (r.status === 400) {
```
Comment (JoshuaVSherman) [inline src/containers/AdminDashboard/AdminController.tsx:313]: // the user's email does not exist in our database, therefore

File: src/containers/AdminDashboard/AdminController.tsx [hunk 3]
```diff
@@ -243,5 +245,84 @@ class AdminController {
...
+      // Create user. Include userType for admin for clc (clc-admin)
+      this.view.setState({ formError: '' });
+      return true;
+    }
+    if (r.body._id && userRoles.indexOf(r.body.userType) === -1) {
+      // if user exists, and isn't an admin, update by user email. Set userType = clc-admin
+      return true;
+    }
+    // If user exists and is already a clc-admin, return error.
```
Comment (JoshuaVSherman) [inline src/containers/AdminDashboard/AdminController.tsx:322]: again, please remove mention of userType here, also search for any other hard-coded userType and remove, those should only be referenced by the commonUtils.getUserType() or process.env.userRoles functions, thanks

**Review by JoshuaVSherman — CHANGES_REQUESTED** [review]

**Review by JoshuaVSherman — APPROVED** [review]

**Pull request merged**